dnf/module/repo_module_dict.py

```diff
@@ -0,0 +1,761 @@
+            raise VersionLockedException(module_spec, module_version.version)
+
+        version_to_lock = module_version.version
+        if list(repo_module.conf.profiles._get()):
+            version_to_lock = module_version.repo_module.conf.version._get()
+        repo_module.lock(version_to_lock)
+
+        if module_form.version and version_to_lock != module_form.version:
+            raise CannotLockVersionException(module_spec, module_form.version,
+                                             "Different version installed.")
+
+        if save_immediately:
+            self.base._module_persistor.commit()
+            self.base._module_persistor.save()
+
+        return module_version.stream, version_to_lock
+
+    def unlock(self, module_spec, save_immediately=False):
+        subj = ModuleSubject(module_spec)
+        module_version, module_form = subj.find_module_version(self)
+
+        repo_module = module_version.repo_module
+
+        if not repo_module.conf.enabled._get():
+            raise EnabledStreamException(module_spec)
+
+        repo_module.unlock()
+
+        if save_immediately:
+            self.base._module_persistor.commit()
+            self.base._module_persistor.save()
+
+        return module_version.stream, module_version.version
+
+    def install(self, module_specs):
+        versions, module_specs = self.get_best_versions(module_specs)
+
+        result = False
+        for module_version, profiles, default_profiles in versions.values():
+            conf = module_version.repo_module.conf
+            if conf.locked._get() and conf.version._get() != module_version.version:
+                logger.warning(module_messages[VERSION_LOCKED]
+                               .format(module_version.name,
+                                       module_version.repo_module.conf.version._get()))
+                continue
+
+            self.enable("{}:{}".format(module_version.name, module_version.stream))
+
+        self.base.sack.reset_includes()
+        self.base.sack.reset_excludes()
+        self.base._setup_excludes_includes()
+
+        for module_version, profiles, default_profiles in versions.values():
+            if module_version.version > module_version.repo_module.conf.version._get():
+                profiles.extend(list(module_version.repo_module.conf.profiles._get()))
+                profiles = list(set(profiles))
+
+            if profiles or default_profiles:
+                result |= module_version.install(profiles, default_profiles)
+
+        if not result and versions and self.base._module_persistor:
+            module_versions = ["{}:{}".format(module_version.name, module_version.stream)
+                               for module_version, profiles, default_profiles in versions.values()]
+            self.base._module_persistor.commit()
+            self.base._module_persistor.save()
+            exit_dnf(module_messages[NOTHING_TO_INSTALL].format(", ".join(module_versions)))
```

Comment:
This is not good way how to end up in script (serious issue)
 - sys.exit() - please remove it
 - Modularity do not use strict option - huge problem from UI
 - Shell command
 - Other requests if skip-broken is used